### ISSUE TITLE
User ability to edit cell data

### ISSUE THREAD

**Issue by akalitenya** [so-epic/kickoff#3]
I watched the product demo videos from Anaplan. Everything has become much clearer. In general, the difference between the Mprove / Looker and the Epic FP&A feature is quite large.

FP&A requires that the user can edit data through a visual interface. Edited data becomes the basis for calculating dependent cells. In Mprove / Looker, the user can do other useful things, but cannot edit the data.

So far, the understanding has come that EpicML cannot be invented by upgrading BlockML, but must be designed from the ground up for its tasks.

**Comment by tgig** [comment]
Thanks for going over everything and making getting us started on these conversations. Very helpful.

I am putting together a more comprehensive plan now and will share with you soon. 

tl;dr: In the early days of our platform, we will likely look a lot more like a BI software (Looker) than a planning software (Anaplan). We are going to ingest and define data like Looker. We are going to use a "model" to connect data (EpicML).  As we grow we probably want to add some collaboration and planning tools on top of the model outputs, so we just need to be sure we're not making fundamental decisions which would prevent that from being possible in the future.

**Comment by tgig** [comment]
But your comment about being able to edit data which then becomes the basis for calculating dependent cells is 100% on point. We will definitely need to be able to do that, and each user should be able to do that in their own "branches" of scenario analysis, then merge them back to the master branch.

**Comment by akalitenya** [comment]
> In the early days of our platform, we will likely look a lot more like a BI software (Looker) than a planning software (Anaplan).

Excellent. As I said earlier, Mprove + BlockML can already do a lot of what Looker + LookML can do. Just continuing to develop this technology, we can save 1-2 years of work (I'm talking about a team of several developers).

> As we grow we probably want to add some collaboration and planning tools on top of the model outputs

Collaboration tools are not directly related to BlockML / LookML, it is Ok.

But planning capability (FP) is a fundamentially different requirenment (user can edit data to alter plans). Therefore, this functionality is not common in BI tools.

It is easy to use, develop and maintain frameworks when they have a narrow specialization.

For me it looks like so if you need to add an FP feature, it will be much easier to write a separate language for this. And this language will not be able to do what BlockML / LookML does, because that is not its purpose.

**Comment by bebelabs** [comment]
https://www.digitalistmag.com/finance/2018/09/13/understanding-intelligent-fpa-dynamic-new-approach-for-finance-06185868/amp

**Comment by akalitenya** [comment]
[](https://www.digitalistmag.com/finance/2018/11/22/replacing-spreadsheets-with-intelligent-fpa-solution-06194315)

"With Excel, calculations are made in reference to their position on a grid. In a modern solution, calculations are named and thus are always applied to the right data."

This is one of the main pain points of Excel users.

**Comment by akalitenya** [comment]
To clarify my point. I think that Epic will eventually be able to do what Looker does and what needed for FP. But to achieve this, it is technically much easier by dividing into functional blocks:
- BI
- FP

These blocks will interact with each other, but technologically will be quite isolated. This makes it easier to research, develop and maintain these modules.

The chain of work with data usually requires the involvement of different skills and, as a result, several different people:
- DevOps Engineer
- Data Engineer
- Data Analyst
- Business User
- Financial Planner

From the point of view of these Epic users, it will also be easier to understand how individual modules work than if you try to mix everything (or a couple of parts) into one language with complex syntax:
- DBT (Data Engineer, Data Analyst)
- BlockML+BI (Data Analyst, Business User)
- EpicML+FP (Financial Planner, Data Analyst, Business User)
- AirFlow (Data Engineer)
- CI / CD (DevOps Engineer)

Business users will enjoy one integrated Epic product (website). They do not care how it is technically implemented.

If in the course of the research work of the language of financial planning the technical understanding changes, then I am not against the unification of languages. But now I do not see the technical benefits of this, but rather only the technical difficulties.

**Comment by bebelabs** [comment]
Great dissection of the various functions and users
